observer/event-store.js

```
@@ -0,0 +1,267 @@
+    const cutoff = now - 60000;
+    while (
+      state.metrics.recentTimestamps.length > 0 &&
+      state.metrics.recentTimestamps[0] < cutoff
+    ) {
+      state.metrics.recentTimestamps.shift();
+    }
+
+    // Update context from envelope fields.
+    // DashboardEmitter._postEvent nests session_id/aiox_agent/aiox_story_id
+    // inside data (see dashboard-emitter.js:320-328). Support both shapes.
+    const session_id = event.session_id || data.session_id;
```

Biome (2.4.6)

[error] This variable is used before its declaration.

(lint/correctness/noInvalidUseBeforeDeclaration)